### PR TITLE
upstream some features --- back merge from intel vpg

### PR DESCRIPTION
**Pull request by xfengcarl**
it includes several changes include ROI, RIR, Processing rate...

### CODE REVIEW

File: va/va.h [hunk 1]
```diff
@@ -2876,6 +2879,7 @@ VAStatus vaQuerySurfaceError(
  * Pre-defined fourcc codes
  */
 #define VA_FOURCC_NV12		0x3231564E
+#define VA_FOURCC_NV21		0x3132564E
```
Comment (yakuizhao) [inline va/va.h]: It will be better that the NV21 definition is added into another patch.

File: va/va.h [hunk 2]
```diff
@@ -2248,6 +2248,8 @@ typedef struct _VAPictureParameterBufferVC1
         } bits;
         unsigned int value;
     } transform_fields;
+    unsigned char luma_scale2;		/* PICTURE_LAYER::LUMSCALE2 */
```
Comment (yakuizhao) [inline va/va.h]: uint8_t is preferred instead of unsigned char.

File: va/va.h [hunk 3]
```diff
@@ -616,6 +629,12 @@ typedef struct _VAConfigAttrib {
 #define VA_RT_FORMAT_RGB32	0x00020000
 /* RGBP covers RGBP and BGRP fourcc */ 
 #define VA_RT_FORMAT_RGBP	0x00100000
+/** 
+ * RGB 10-bit packed format with upper 2 bits as alpha channel. 
+ * The existing pre-defined fourcc codes can be used to signal 
+ * the position of each component for this RT format.  
+ */
+#define VA_RT_FORMAT_RGB32_10BPP 0x00200000
```
Comment (yakuizhao) [inline va/va.h]: This can be defined in another patch.

**Comment by xfengcarl** [comment]
after rebase, all commits changes.

File: va/va.h [hunk 4]
```diff
@@ -2351,6 +2352,8 @@ typedef struct _VAPictureParameterBufferH264
         unsigned int value;
     } pic_fields;
     unsigned short frame_num;
+    unsigned char num_ref_idx_l0_default_active_minus1;
+    unsigned char num_ref_idx_l1_default_active_minus1;
```
Comment (xhaihao) [inline va/va.h]: The two flags are added for slice decoding base mode (VA_DEC_SLICE_MODE_BASE). It would be better to add some comments here.
